processing/src/test/java/io/druid/segment/IndexMergerTest.java

```diff
@@ -813,6 +817,46 @@ public void testDisjointDimMerge() throws Exception
     Assert.assertArrayEquals(new Object[]{1L}, boatList.get(4).getMetrics());
   }
 
+  @Test
+  public void testMismatchedMetrics() throws IOException
+  {
+    IncrementalIndex index1 = IncrementalIndexTest.createIndex(new AggregatorFactory[]{
```

Comment:
IncrementalIndex should be closed

I put in `io.druid.segment.CloserRule` for just such occasions, where you can specify a `@Rule CloserRule testCloser` and just `testCloser.closeLater(index1)`